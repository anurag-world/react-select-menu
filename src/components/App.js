// Imports
import React, { Component } from 'react';
import Menu from './simple/Menu';
import MenuExt from './extended/MenuExt';

class Main extends Component {
  constructor(props) {
    super(props);

    // Initializing the state & data
    this.state = {
      // Dark Mode
      darkMode: false,
    };
  }

  // Toggle Dark Mode
  toggleDarkMode = () => {
    this.setState((state) => ({
      darkMode: !state.darkMode,
    }));
  };

  render() {
    return (
      <div
        className="container"
        data-theme={this.state.darkMode ? 'dark' : 'light'}
      >
        <div>
          <button className="dark-mode-btn" onClick={this.toggleDarkMode}>
            {this.state.darkMode
              ? 'Switch to Light Mode'
              : 'Switch to Dark Mode'}
          </button>
        </div>

        <div className="select-menu">
          {/* Simple Menu */}
          <div className="simple-menu">
            <h4 style={{ marginBottom: 20 }}>Simple Menu: </h4>
            <Menu />
          </div>

          {/* Extended Menu */}
          <div>
            <h4 style={{ marginBottom: 20 }}>Extended Menu: </h4>
            <MenuExt />
          </div>
        </div>
      </div>
    );
  }
}
export default Main;
