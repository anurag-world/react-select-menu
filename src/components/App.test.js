/**
 * @jest-environment jsdom
 */

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import Data from '../data/list.json';
import DataExt from '../data/listExt.json';

test('App Snapshot Test', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
  ReactDOM.render(
    <React.StrictMode>
      <App />
    </React.StrictMode>,
    document.createElement('root')
  );
});

// Data Check
test('Data is available for the list of simple menu items', () => {
  expect(Data).toEqual(expect.arrayContaining(Data));
});

test('Data is available for the list of extended menu items', () => {
  expect(DataExt).toEqual(expect.arrayContaining(DataExt));
});
