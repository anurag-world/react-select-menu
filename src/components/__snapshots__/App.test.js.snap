// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot Test 1`] = `
<div
  className="container"
  data-theme="light"
>
  <div>
    <button
      className="dark-mode-btn"
      onClick={[Function]}
    >
      Switch to Dark Mode
    </button>
  </div>
  <div
    className="select-menu"
  >
    <div
      className="simple-menu"
    >
      <h4
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        Simple Menu: 
      </h4>
      <button
        className="menu-header"
        onClick={[Function]}
      >
        Hopper
        <span
          aria-hidden={true}
          className="fa fa-angle-down icon-color"
          style={
            Object {
              "marginLeft": 12,
            }
          }
        />
      </button>
    </div>
    <div>
      <h4
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        Extended Menu: 
      </h4>
      <button
        className="menu-header"
        onClick={[Function]}
      >
        Hopper
        <span
          aria-hidden={true}
          className="fa fa-angle-down icon-color"
          style={
            Object {
              "marginLeft": 12,
            }
          }
        />
      </button>
    </div>
  </div>
</div>
`;
